src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/DefaultRulesValidationSpec.groovy

```diff
@@ -41,7 +55,50 @@ class DefaultRulesValidationSpec extends HealthCheckSpecification {
             rules.proper.sort() == sw.defaultCookies.sort()
         }
 
-        where: "Run for all unique switches"
-        sw << getTopology().getActiveSwitches().unique { activeSw -> activeSw.description }
+        cleanup: "Restore original switch props"
+        switchHelper.updateSwitchProperties(sw, originalProps)
+
+        where: "Run for all combinations of unique switches and switch modes"
+        [swProps, sw] <<
+                [[
+                    [
+                        multiTable: false,
+                        switchLldp: false,
+                        switchArp: false
+                    ],
+
+                    [
+                        multiTable: true,
+                        switchLldp: false,
+                        switchArp: false
+                    ],
+                    [
+                        multiTable: true,
+                        switchLldp: true,
+                        switchArp: true
+                    ]
+                ], getTopology().getActiveSwitches().unique { activeSw -> activeSw.description }
+                ].combinations().findAll { Map swProp, Switch _sw ->
+                    //filter out combinations where we pick non-multitable switch and do multitable switch props
+                    !(swProp.multiTable && !_sw.features.contains(SwitchFeature.MULTI_TABLE))
+                }
+        propsDescr = getDescr(swProps.multiTable, swProps.switchLldp, swProps.switchArp)
+    }
+
+    String getDescr(boolean multitable, boolean lldp, boolean arp) {
```

Comment:
Any arguments?

Reply:
All passed parameters can be taken the same object. 
So, I strongly believe it is easier to pass an object. 
In addition you don't need to remember the correct order of passed arguments.

Reply:
My thought process is: Your option is 'not safe'. You have to 'remember' the key names and do not get any suggestions from the IDE. It is not type-safe as well. I prefer not to spread dynamic code across the place and get rid of it as soon as possible, since any typo will only be caught during runtime.
I understand that we pass the same map to the test itself, but I would happily use custom classes(like `ConnectedDeviceTestData` in ConnectedDevicesSpec) if it didn't require additional coding. And I would be happy to pass a strongly-typed object here as a param. I would also may have considered using map if we had too many params to pass(but its only 3 here).
So you gave only 1 argument: ' you don't need remember the order of passed arguments'. Nobody have to 'remember' it, it is the method signature, order will always be suggested by the IDE..unlike remembering keys of your map.
Don't get me wrong, I see your suggestion as an option and don't think its bad, but unless you give more arguments I don't see real reasons to change current code. It's just different